apps/dashboard/shared/hooks/graphql-client/useVotingPower.ts

```diff
@@ -0,0 +1,342 @@
+
+  // Main data query
+  const {
+    data: delegatorsVotingPowerDetails,
+    loading,
+    error,
+    refetch,
+    fetchMore,
+    networkStatus,
+  } = useGetDelegatorVotingPowerDetailsQuery({
+    context: {
+      headers: {
+        "anticapture-dao-id": daoId,
+      },
+    },
+    variables: {
+      address,
+      after: undefined,
+      before: undefined,
+      orderBy,
+      orderDirection,
+      limit: itemsPerPage,
+    },
+    notifyOnNetworkStatusChange: true,
+    fetchPolicy: "cache-and-network",
+  });
+
+  // Count query
+  const { data: countingData } = useGetVotingPowerCountingQuery({
+    variables: {
+      address,
+    },
+    context: {
+      headers: {
+        "anticapture-dao-id": daoId,
+      },
+    },
+  });
+
+  // Refetch data when sorting changes to ensure we start from page 1
+  useEffect(() => {
+    refetch({
+      after: undefined,
+      before: undefined,
+      orderBy,
+      orderDirection,
+    });
+  }, [orderBy, orderDirection, refetch]);
+
+  const accountBalances = delegatorsVotingPowerDetails?.accountBalances?.items;
+
+  // ------------------------------------------------------------------
+  // Prepare the array of delegator addresses once balances are fetched
+  // ------------------------------------------------------------------
+  const delegatorAddresses: string[] = accountBalances
+    ? accountBalances.map((item) => item.accountId)
+    : [];
+
+  // ------------------------------------------------------------------
+  // Fetch delegation timestamps (skipped until we have delegatorAddresses)
+  // ------------------------------------------------------------------
+  const {
+    data: delegationsTimestampData,
+    loading: tsLoading,
+    error: tsError,
+  } = useGetDelegationsTimestampQuery({
+    context: {
+      headers: {
+        "anticapture-dao-id": daoId,
+      },
+    },
+    variables: {
+      delegate: address,
+      delegator: delegatorAddresses,
+      daoId: daoId,
+    },
+    skip: delegatorAddresses.length === 0,
+  });
+
+  // ------------------------------------------------------------------
+  // Build timestamp lookup <delegatorAccountId> -> timestamp
+  // ------------------------------------------------------------------
+  const timestampMap: Record<string, string | number | undefined> =
+    Object.fromEntries(
+      (delegationsTimestampData?.delegations.items || []).map((delegation) => [
+        delegation.delegatorAccountId?.toLowerCase(),
+        delegation.timestamp,
+      ]),
+    );
+
+  // ------------------------------------------------------------------
+  // Enrich balances with timestamp (fallback to undefined)
+  // ------------------------------------------------------------------
+  const balancesWithTimestamp = (accountBalances || []).map((account) => ({
+    ...account,
+    timestamp: timestampMap[account.accountId.toLowerCase()],
+  }));
+
+  const { data: top5Delegators } = useGetTop5DelegatorsQuery({
```

Comment:
I'm not a big fan of having a number here, can't you change to "useGetTopFiveDelegatorsQuery" and "topFiveDelegators" or just "useGetTopDelegatorsQuery" and "topDelegators"?